signer/tests/integration/withdrawal_reject.rs

```diff
@@ -603,3 +604,135 @@ async fn reject_withdrawal_validation_request_being_fulfilled() {
 
     testing::storage::drop_db(db).await;
 }
+
+/// For this test we check that the `RejectWithdrawalV1::validate` function
+/// returns a withdrawal validation error with a RequestStillActive
+/// message when the database indicates that it is possible that the
+/// withdrawal request is being fulfilled by a sweep transaction in the
+/// mempool.
+#[tokio::test]
+async fn reject_withdrawal_validation_request_still_active() {
+    // Normal: this generates the blockchain as well as a transaction
+    // sweeping out the funds for a withdrawal request. This is just setup
+    // and should be essentially the same between tests.
+    let db = testing::storage::new_test_database().await;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+    let (rpc, faucet) = regtest::initialize_blockchain();
+
+    let amount = 1_000_000;
+    let signers = TestSignerSet::new(&mut rng);
+    let amounts = [
+        SweepAmounts {
+            amount,
+            max_fee: amount / 2,
+            is_deposit: true,
+        },
+        SweepAmounts {
+            amount,
+            max_fee: amount / 2,
+            is_deposit: false,
+        },
+    ];
+
+    let mut setup = TestSweepSetup2::new_setup(signers, faucet, &amounts);
+
+    let mut ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_first_bitcoin_core_client()
+        .with_mocked_stacks_client()
+        .with_mocked_emily_client()
+        .build();
+
+    // Normal: the request has not been marked as completed in the smart
+    // contract.
+    set_withdrawal_incomplete(&mut ctx).await;
+
+    // Normal: the signer follows the bitcoin blockchain and event observer
+    // should be getting new block events from bitcoin-core. We haven't
+    // hooked up our block observer, so we need to manually update the
+    // database with new bitcoin block headers.
+    fetch_canonical_bitcoin_blockchain(&db, rpc).await;
+
+    // Different: we need to store a row in the dkg_shares table so that we
+    // have a record of the scriptPubKey that the signers control. We need
+    // this so that the donation gets picked up correctly below.
+    setup.store_dkg_shares(&db).await;
```

Comment:
Why is this _Different_?

Reply:
It isn't. I copied all of the comments from another test, and didn't update anything yet. Also, it isn't exactly "Different" in that other test either.

Reply:
I updated this in https://github.com/stacks-network/sbtc/pull/1411/commits/55a0aae4a01e50369815c6558035524368cc6b30